File: server/src/protocol/SensorTagSamplesCommand.cpp
#include "SensorTagSamplesCommand.h"
#include "CommandVisitor.h"

using namespace std;
using namespace communication;

SensorTagSamplesCommand::SensorTagSamplesCommand()
        : Command(10, CommandType::SENSOR_TAG_SAMPLES)
{}

SensorTagSamplesCommand::~SensorTagSamplesCommand()
{}

vector<uint8_t> SensorTagSamplesCommand::getFrameBytes()
{
    initializeDataSize();

    vector<uint8_t > frame = getHeader();
    frame.push_back(static_cast<uint8_t>(commandType_));

    return frame;
}

std::string SensorTagSamplesCommand::getName()
{
    return string("SensorTagSamplesCommand");
}

void SensorTagSamplesCommand::accept(CommandVisitor &visitor)
{
    visitor.visit(*this);
}

void SensorTagSamplesCommand::initializeDataSize()
{
    uint16_t dataSize = sizeof(commandType_);

    setDataSize(dataSize);
}

File: mobile_client/app/src/main/java/wpam/mobile_client/protocol/SensorTagSamplesCommand.java
package wpam.mobile_client.protocol;

import java.util.ArrayList;

public class SensorTagSamplesCommand extends Command
{
    public SensorTagSamplesCommand()
    {
        super(10, CommandType.SENSOR_TAG_SAMPLES);
    }
    public final void close()
    {
        super.close();
    }

    @Override
    public ArrayList<Integer> getFrameBytes()
    {
        initializeDataSize();

        ArrayList<Integer> frame = getHeader();
        frame.add(commandType_.getValue());

        return frame;
    }
    @Override
    public String getName()
    {
        return new String("SensorTagSamplesCommand");
    }

    @Override
    public void accept(CommandVisitor visitor)
    {
        visitor.visit(this);
    }

    @Override
    protected void initializeDataSize()
    {
        setDataSize(1);
    }
}